### ISSUE TITLE
ttnn.matmul doesn't support broadcast of batch dims of input tensor a

### ISSUE THREAD

**Issue by vigneshkeerthivasanx** [tenstorrent/tt-metal#22134]
### Describe the bug

TTNN matmul op does not work when the batch dim of input_tensor_a needs to be broadcasted.

### Steps to reproduce the issue

To Reproduce:

import ttnn
import torch
 
def test_matmul_broadcast(device):
    x = torch.randn(1, 1, 1, 1, 3, 3)
    y = torch.randn(1, 4, 160, 160, 3, 1)
    golden = torch.matmul(x, y)
 
    tt_x = ttnn.from_torch(x, layout=ttnn.TILE_LAYOUT, device=device)
    tt_y = ttnn.from_torch(y, layout=ttnn.TILE_LAYOUT, device=device)
    tt_out = ttnn.matmul(tt_x, tt_y)
    tt_out = ttnn.to_torch(tt_out)
 
    assert ttnn.pearson_correlation_coefficient(golden, tt_out) >= 0.99

Error encountered after executing the above test:

**Comment by vigneshkeerthivasanx** [comment]
cc: @saichandax @mbahnasTT

**Comment by vigneshkeerthivasanx** [comment]
The issue can be resolved with a workaround using `ttnn.expand` and then using `ttnn.matmul`, hence closing this issue.

**Comment by moritztng** [comment]
But the workaround with ttnn.expand uses more memory and is slow right? I used ttnn.repeat in my project tt-boltz and it's too slow.
